File: Year-2023/LeetCode-June-Challenge/FindtheHighestAltitude.js
var largestAltitude = function(gain) {
    let res = 0, temp = 0;
    for(let x of gain){
        temp += x;
        res = Math.max(res, temp);
    }
    return res;
};

let gain = [-5,1,5,0,-7];
console.log(largestAltitude(gain));

File: Year-2023/LeetCode-June-Challenge/FindtheHighestAltitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int res = 0, temp = 0;
        for(int x: gain){
            temp += x;
            res = Math.max(res, temp);
        }
        return res;
    }
}
public class FindtheHighestAltitude{
    public static void main(String[] args) {
        int[] gain = {-5,1,5,0,-7};
        System.out.println(new Solution().largestAltitude(gain));
    }
}